samza-core/src/main/scala/org/apache/samza/system/SystemConsumersMetrics.scala

```diff
@@ -19,12 +19,13 @@
 
 package org.apache.samza.system
 
-import org.apache.samza.metrics.MetricsRegistry
 import org.apache.samza.metrics.MetricsRegistryMap
 import org.apache.samza.metrics.Counter
 import org.apache.samza.metrics.MetricsHelper
+import org.apache.samza.metrics.ReadableMetricsRegistry
 
-class SystemConsumersMetrics(val registry: MetricsRegistry = new MetricsRegistryMap) extends MetricsHelper {
+class SystemConsumersMetrics(val registry: ReadableMetricsRegistry = new MetricsRegistryMap,
+  val prefix: String = "") extends MetricsHelper {
```

Comment:
Can we call it 'source' instead? Or are the semantics very different?

Reply:
We discussed offline, source is a TaskInstance metrics specific thing and should be removed. 'prefix' has similar semantics and is the better way to do this. @rmatharu please add the ticket here with the discussion summary.

Reply:
https://issues.apache.org/jira/browse/SAMZA-2125